### PR TITLE
[MAJOR] Add new optional paramter to pbxProject.addTarget

### PR DESCRIPTION
**Pull request by NiklasMerz**
This allows plugins to set bundle ids for extensions.

Necessary with cordova-ios >= 5 after the introduction of $(PRODUCT_NAME) in https://github.com/apache/cordova-ios/pull/409/files. Plugins with extension now need to set their bundleId to make them work in Xcode.

### CODE REVIEW

**Comment by brodybits** [comment]
Any news on this work or the issue on iOS?

**Comment by NiklasMerz** [comment]
I need this for hooks in my plugin that needs to change the targets in the Xcode project. See https://github.com/GEDYSIntraWare/cordova-plugin-call-directory/commit/9ad709a6568a7edcca460a60019d7896696beafe

File: test/addTarget.js [hunk 1]
```diff
@@ -93,6 +94,25 @@ exports.addTarget = {
...
+        test.ok(target.pbxNativeTarget.productName);
+        test.ok(target.pbxNativeTarget.productReference);
+        test.ok(target.pbxNativeTarget.productType);
+        test.ok(target.pbxNativeTarget.buildConfigurationList);
+        test.ok(target.pbxNativeTarget.buildPhases);
+        test.ok(target.pbxNativeTarget.buildRules);
+        test.ok(target.pbxNativeTarget.dependencies);
+  
+        test.done();
+    }
```
Comment (timbru31) [inline test/addTarget.js]: Missing comma, therefore the CI failure

Comment (NiklasMerz) [reply test/addTarget.js]: Thank you. I messed up the merge.

Comment (timbru31) [reply test/addTarget.js]: No worries, happens :)

**Comment by timbru31** [comment]
Should we aim to merge the PR, even if apache/cordova-ios#538 is not yet fixed?

**Comment by NiklasMerz** [comment]
I think yes. Some plugins need this package as a dependency for hooks that modify only one target. So getting this in the upcoming release would be good.

I cannot get my head around how to fix https://github.com/apache/cordova-ios/issues/538 properly but this is definitly an issue and I hacked around it. Cordova iOS definityl causes issues for the tiny percentage of apps that use more than the default target (watch, share, calldirectory extensions). But we should discuss this in the issue.

**Comment by NiklasMerz** [comment]
Works together with #106 and https://github.com/apache/cordova-ios/pull/820 to fix https://github.com/apache/cordova-ios/issues/538

**Pull request merged**